### ISSUE TITLE
Objectifs pour la V1 ? CRAN ?

### ISSUE THREAD

**Issue by ThomasSoeiro** [SNDStoolers/sndsTools#33]
Quels sont nos objectifs avant de publier une V1 (ou V diffusable du moins) ? On peut faire un [milestone sur GitHub](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) pour suivre l'avancée et prioriser.

Est-ce qu'on vise une publication sur le CRAN ? Ca ne me semble pas indispensable si la Cnam accepte de l'installer sur le portail sans qu'il soit sur le CRAN (car le paquet ne peut servir que sur le portail).

**Comment by strayMat** [comment]
Je suis d'accord que le cran ne me semble pas indispensable (du moins pour une v1). 

Ce serait déjà cool d'avoir les 5 fonctions d'extraction déjà existantes avec :

- arguments cohérents et test des arguments sur l'ensemble des fonctions : https://github.com/strayMat/sndsTools/issues/20
- documentation, exemples et tests : https://github.com/strayMat/sndsTools/issues/1 et  https://github.com/strayMat/sndsTools/issues/17
- la gestion des identifiants : https://github.com/strayMat/sndsTools/issues/28

Côté dépôt, je vois : 
- le guide de contribution https://github.com/strayMat/sndsTools/issues/31
- ajouter  une (vague) roadmap ??? 
- peut être le devtools::check et le linter (à discuter, surtout si on ne vise pas le cran)

Est-ce que vous voyez autre chose ? Une milestone github serait pas mal effectivement une fois qu'on est à peu près d'accord de ce qu'on y inclus.

**Comment by ThomasSoeiro** [comment]
> Je suis d'accord que le cran ne me semble pas indispensable (du moins pour une v1).

Faudra qu'on demande à la Cnam s'ils sont d'accord pour installer un paquet spécifique SNDS qui n'est pas sur le CRAN ou s'il veulent qu'on le publie. Faudra voir s'ils peuvent nous proposer un processus de MAJ ad hoc vu qu'on va forcément découvrir des boulettes une fois installer pour pas laisser un paquet buggué à disposition des utilisateurs.

**Comment by abelloir-zz** [comment]
Il me semble que des fonctions utilitaires SAS ont déjà été mises à disposition des utilisateurs, on pourrait demander quelque chose de similaire :)

**Comment by strayMat** [comment]
C'est sûr qu'il faudra un processus de mis-à-jour. Idéalement ils ont quelqu'un qui peut faire une maj du paquet tous les X mois ou à terme quand il y a une release ? 
Mais c'est vrai qu'au début, ça peut juste être la mise à disposition des scripts dans un dossier.

**Comment by ThomasSoeiro** [comment]
Un moyen de mettre à disposition les fonctions des utilisateurs du portail (si ça coince pour mettre à disposition le paquet) :

De notre côté :
- à partir d'un Unix : `cat sndsTools/R/* > sndsTools.R` (NB : cas d'usage où il faut bien veiller à avoir un [retour à la ligne à la fin des fichiers](https://github.com/strayMat/sndsTools/pull/12#discussion_r1734971983))
- partager le `sndsTools.R`

Côté des utilisateurs :
- importer le fichier `sndsTools.R` sur le portail.
- dans R : `source("sndsTools.R")`

(PS : ça marche pour tout paquet composé exclusivement en R et dont on a besoin sur le portail mais qui n'est pas installé 😉 )

**Comment by strayMat** [comment]
Clairement une astuce à mettre dans la future vignette usage.Rmd ou installation.Rmd

**Comment by ThomasSoeiro** [comment]
J'avoue ne pas me servir du paquet et pas avoir testé donc je ne sais pas où on en est précisément... 😬 Est-ce que vous pensez qu'on a suffisamment avancé pour une première publication ? (qui entrainera logiquement communication plus large + appel à contribution). Ca me semble également judicieux pour éviter qu'on s'essouffle (même si le rôle de @abelloir-zz et désormais de @strayMat comme locomotive est remarquable 🚀 )

Du coup ça nous renvoie à https://github.com/SNDStoolers/sndsTools/issues/33#issuecomment-2326904146 :

> Faudra qu'on demande à la Cnam s'ils sont d'accord pour installer un paquet spécifique SNDS qui n'est pas sur le CRAN ou s'il veulent qu'on le publie. Faudra voir s'ils peuvent nous proposer un processus de MAJ ad hoc vu qu'on va forcément découvrir des boulettes une fois installer pour pas laisser un paquet buggué à disposition des utilisateurs.

Je peux m'en occuper.

PS : Je demande parce que j'ai un cas d'usage pour lequel j'aimerais bien que le paquet soit dispo sur le portail : des collègues peu initiés au SNDS ont besoin d'une petite extraction du PMSI MCO sur notre région. Je pense qu'ils pourraient être autonome avec le paquet.

**Comment by strayMat** [comment]
Je pense qu'on est pas loin : 

J'ai retracé ce qu'il restait à faire dans ce milestone : https://github.com/SNDStoolers/sndsTools/milestone/1
En résumé il reste : 
- la fonction hospital_stays : https://github.com/SNDStoolers/sndsTools/pull/47 c'est à moi d'avancer, je pense que j'aurais fini la PR avant fin février. 
- la gestion des identifiants : https://github.com/SNDStoolers/sndsTools/pull/48 j'ai l'impression qu'on y est presque. 

Je propose de remettre à une V2 les tests car y a encore de la discussion et pas mal de travail je pense.
